rewrite-core/src/main/java/org/openrewrite/GitRemote.java

```diff
@@ -142,12 +143,12 @@ public URI buildUri(Service service, String origin, String path, String protocol
                                 .anyMatch(o -> o.equalsIgnoreCase(stripProtocol(origin)))
                         )
                         .flatMap(server -> server.getUris().stream())
-                        .filter(uri -> uri.getScheme().equals(protocol))
+                        .filter(uri -> Parser.normalize(uri).getScheme().equals(protocol))
                         .findFirst()
                         .orElseGet(() -> {
```

Comment:
You could add a map to normalize the uri to remove duplicate calls to do so

Reply:
There is no duplicate call. I do not want the normalized URI as output, just in the filter. And if none are found then we normalize something else

Reply:
Ah, sorry, I didn't read the `orElseGet`